docs/DesignDocs/SGX_Endorsements_V2.md

```diff
@@ -0,0 +1,222 @@
+
+### Get QE Identity Info
+1. The 'qeIdentity' block was renamed to 'enclaveIdentity'.
+2. A new field, enclaveIdentity:id was added to differentiate QE Id info from Quoting Enclave (QE) and Quoting Validation Enclave (QVE).
+3. enclaveIdentity:version changed from 1 to 2.
+4. New field enclaveIdentity:tcbEvaluationDataNumber.
+5. qeIdentity:isvsvn moved to enclaveIdentity:tcbLevels:tcb:isvsvn.
+6. New fields in enclaveIdentity:tcbLevels named tcbData, tcbStatus and advisoryIDs.
+
+### New field definitions
+
+#### tcbType:
+	type: integer
+	example: 0
+	description: >- Type of TCB level composition that determines TCB level comparison logic
+
+##### tcbEvaluationDataNumber:
+	type: integer
+	example: 2
+	description: >- A monotonically increasing sequence number changed when Intel updates the content of the TCB evaluation data set: TCB Info, QE Idenity and QVE Identity. The tcbEvaluationDataNumber update is synchronized across TCB Info for all flavors of SGX CPUs (Family-Model-Stepping-Platform-CustomSKU) and QE/QVE Identity. This sequence number allows users to easily determine when a particular TCB Info/QE Idenity/QVE Identiy superseedes another TCB Info/QE Identity/QVE Identity (value: current TCB Recovery event number stored in the database).
+
+##### tcbDate:
+	type: string
+	format: date-time
+	description: >- Representation of date and time when the TCB level was certified not to be vulnerable to any issues described in SAs that were published on or prior to this date. The time shall be in UTC and the encoding shall be compliant to ISO 8601 standard (YYYY-MM-DDThh:mm:ssZ).
+
+##### tcbStatus:
+	type: string
+	description: TCB level status
+
+##### advisoryIDs:
+	type: array
+	description: >- Array of Advisory IDs describing vulnerabilities that this TCB level is vulnerable to. Note: The value can be different for different FMSPCs. This field is optional. It will be present only if the list of Advisory IDs is not empty.
+	items:
+        type: string
+
+For more information on the new fields please see [API Version 2](https://api.portal.trustedservices.intel.com/documentation#pcs-certificate-v2).
+
+
+## Changes
+
+### Update current structs with new additional fields and create new type definitions.
+
+```C
+typedef enum _oe_tcb_level_status
+{
+    // Existing fields
+    OE_TCB_LEVEL_STATUS_UNKNOWN,
+    OE_TCB_LEVEL_STATUS_REVOKED,
+    OE_TCB_LEVEL_STATUS_OUT_OF_DATE,
+    OE_TCB_LEVEL_STATUS_CONFIGURATION_NEEDED,
+    OE_TCB_LEVEL_STATUS_UP_TO_DATE,
+
+    // New field for new value "OutOfDateConfigurationNeeded"
+    OE_TCB_LEVEL_STATUS_OUTOFDATE_CONFIGURATION_NEEDED,
+
+    __OE_TCB_LEVEL_MAX = OE_ENUM_MAX,
+
+} oe_tcb_level_status_t;
+
+// Existing TCB Level struct for TCB Info.  Will rename this
+// struct to oe_tcb_tcb_level given that
+// QE identity info also has its TCB level data field.
+typedef struct _oe_tcb_level
+{
+    // Existing fields
+    uint8_t sgx_tcb_comp_svn[16];
```

Comment:
This looks like an sgx-specific field in a non-sgx-specific struct.  Either make the field name be generic, or make the struct name have sgx in the name.

Reply:
This existing struct is defined in the common/sgx/tcbinfo.h file which implies sgx.  I think it makes sense to rename it sgx specific, but if we do this it should be part of a different issue given that all the types under common/sgx are sgx specific.